README.md
# Installation Scripts for Ubiquiti's UCRM - Complete ISP Management Platform 

This repo includes installation and update scripts for UCRM system. UCRM is a streamlined billing and customer management system with credit card processing and many automatic features, such as recurring invoicing, automatic suspension of late payers, etc. Enjoy easy and customizable billing platform, simply set up your clients, services, online payment gateways and let UCRM help you grow your business. For advanced network management, go to [UNMS](https://unms.ubnt.com).

[Installation](https://help.ubnt.com/hc/en-us/articles/115000994708-UCRM-Installation-Guide) | [Knowledge Base](https://help.ubnt.com/hc/en-us/sections/204958307-UCRM-WISP-Management-Platform) | [Community](https://community.ubnt.com/t5/UCRM-Complete-WISP-Management/bd-p/UCRM) | [Changelog](https://github.com/Ubiquiti-App/UCRM/blob/master/CHANGELOG.md) | [Roadmap](https://ucrm.ubnt.com/)


CHANGELOG.md
# Changelog

## 2.16.15 (2021-03-09)

### Fixed
*   Fixed possible wrong client balance calculation after unvoiding proforma invoice.
*   Fixed issues with Stripe ACH payments
*   Fixed issues with netflow feature not measuring any data.
*   Fixed FCC report issues when the company name contains "/"
*   Possible fix for IMAP import for some strange mail servers and email formats.

### Added
*   Cookies security improvement.

## 2.16.14 (2020-06-23)

### Fixed
*   Fixed issues when having multiple Authorize.Net subscriptions for a single client.
*   Fixed rare issues with invoicing of services having broken suspend/unsuspend process.
*   Fixed several XSS vulnerabilities.

## 2.16.13 (2020-05-20)

### Fixed
*   Fixed payment form, showing to the client, that the invoices is already paid when attempted to pay it (just in some rare cases).
*   Fixed attachments for client zone contact form.
*   Fixed wrong due days count shown in invoice grid in some rare cases related to leap years.
*   Fixed MapBox API deprecations.

## 2.16.12 (2020-03-23)

### Changed
*   Better handling of zero amount proformas and voiding invoices that were created from proformas. (The related proforma invoice is voided / restored automatically when the invoice is voided / restored manually)
*   Invoice draft API - editing draft is now enabled.

### Fixed
*   Fixed failing very long PDF generation (e.g. while creating huge invoice count into PDF)
*   Fixed client edit crash in some rare cases.
*   Fixed invoice grid sorting.
*   Fix crash when voiding processed proforma invoice via API.
*   Minor rounding error for online payments views.
*   Fixed invoicing issues with services having deferred change set or executed.
*   Fixed rare crash on the service detail page.
*   Fixed near due date notification offset. Now, when set to 1 for example, notification is sent 1 day before the due date as expected.
*   Fixed ticketing IMAP import with strange encoding in email subject.
*   Fixed client zone account statement - proforma invoices were not marked as proformas.
*   Fixed minor UX issues and bugs.

## 2.16.11 (2020-01-30)

### Changed
*	Allowed custom CSS for some more elements (forms, scripts, iframes, objects)

### Fixed
*	Fixed print buttons that did nothing in some browsers.
*	Fixed bounced emails causing ticketing loop in some cases.
*	Fixed rare issues in system log.
*	CSV structure updated for the FCC export.
*	Fixes for some webhook events.
*	Fixed bug in invoice draft edit - use credit toggle had no effect.
*	Fixed invoice not created on specific date (this might happen in some rare cases with period start on 28th).
*	Fixed filtering view for failed outgoing emails.

## 2.16.10 (2019-12-16)

### Changed
*   PHP upgraded to 7.3.12

### Fixed
*	Improved MercadoPago payments for amounts and currencies not having cents.
*	Fixed issues with manual service suspension button.
*	Fixed rare issues with pay online button for Stripe gateway.
*	Fixed UCRM being stuck in some rare cases due to Ticketing bundle issues.
*	Other minor fixes.

## 2.16.9 (2019-11-19)

### Changed
* PHP upgraded to 7.3.11

### Fixed
*	The same name for a custom attribute for different entities (client, service, invoice) can now be created.
*	Fixed template placeholders (having "_" in their name) not showing any value.
*	Fixed wrong invoice issue date in rare cases (only when invoicing 2 or more periods). The invoice was issued one day ahead.
*	Improved rare situation when a paid proforma is created (credit is applied), the resulting invoice is now issued immediately, together with that proforma invoice.
*	Fixed CSV payment import - matching to a custom payment method names fixed.
*	API payment to invoice draft is not allowed now, fixed other minor bugs in payment API.
*	Fixed issues with empty datetime parameter in plugin configuration.
*	Other minor fixes.

## 2.16.8 (2019-09-27)
### Changed
*	PHP upgraded to 7.2.23

### Fixed
*	Fixed rare crash on client detail after subscription is re-created.
*	Fixed failing invoice number generator when approving invoice drafts (in some rare cases).
*	Service reactivation is now forbidden for archived clients.
*   Fixed failing plugin execution (manual and automatic).
*	Minor fixes.

## 3.0.4 (in UNMS v1.0.4) (2019-09-25)
### Changed
*	PHP upgraded to v7.3.

### Fixed
*	Improved payment page shown when clicked on "pay online button" for already paid invoice.
*	Optimized disk usage when restoring backup of UCRM v2 in UNMS.
*	Better UX for new subscription form in client zone.
*	API improvements, better validations and fixes.
*	Fixed redirect on the service reactivation page for clients.
*	Fixed rare crash on client detail after subscription is re-created.
*	Fixed failing invoice number generator when approving invoice drafts (in some rare cases).
*	Minor fixes.

## 2.16.7 (2019-09-16)
### Changed
* PHP upgraded to 7.2.22
### Fixed
*	Fixed rounding issue in tax recapitulation on invoice overview / PDF.
*	Fixed IPpay payments, error 900 caused by breaking changes on IPpay side.
*	Fixed rounding for some rare cases of IPpay payments.
*	CRM webhook logs are pruned periodically now.
*	Fixed issues with saving the plugin's configuration.
*	Improved stripe payment processing and shown metadata of related crm payment.
*	Improved payment page shown when clicked on "pay online button" for already paid invoice.

## 3.0.3 (in UNMS v1.0.3) (2019-09-12)
### Added
*   Added JOD currency.
*   Now, it's possible to manually invoice service for more than 3 years ahead.
*   Improved stripe payment processing, also metadata of related CRM payments are now displayed.

### Fixed
*	Line breaks in the default ticketing notification template now works correctly.
*	Fixed issues with two or more CRM organizations using the same Stripe account.
*	Placing images into email templates now works without issues. Guides and template variables doc were improved.
*	CRM webhook logs are now pruned periodically.

uninstall.sh
#!/usr/bin/env bash

set -o errexit
set -o errtrace
set -o nounset
set -o pipefail
#set -o xtrace

UCRM_PATH="${UCRM_PATH:-}"
if [[ "${UCRM_PATH}" = "" ]] && [[ "${BASH_SOURCE+x}" = "x" ]]; then
    UCRM_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" > /dev/null && pwd)"
fi
if [[ "${UCRM_PATH}" = "" ]]; then
    UCRM_PATH="."
fi

do_uninstall() {
    if (which docker > /dev/null 2>&1); then
        echo "Removing UCRM docker containers and images."

        if [[ -f "${UCRM_PATH}/docker-compose.yml" ]] && [[ -f "${UCRM_PATH}/docker-compose.migrate.yml" ]]; then
            if [[ -f "${UCRM_PATH}/docker-compose.version.yml" ]]; then
                docker-compose -f "${UCRM_PATH}/docker-compose.yml" -f "${UCRM_PATH}/docker-compose.migrate.yml" -f "${UCRM_PATH}/docker-compose.version.yml" stop
                docker-compose -f "${UCRM_PATH}/docker-compose.yml" -f "${UCRM_PATH}/docker-compose.migrate.yml" -f "${UCRM_PATH}/docker-compose.version.yml" rm -af
            else
                docker-compose -f "${UCRM_PATH}/docker-compose.yml" -f "${UCRM_PATH}/docker-compose.migrate.yml" stop
                docker-compose -f "${UCRM_PATH}/docker-compose.yml" -f "${UCRM_PATH}/docker-compose.migrate.yml" rm -af
            fi
        fi

        local oldImages
        oldImages=$(docker images -a | grep "^ubnt/ucrm-billing" | awk '{print $3}') || true
        if [[ "${oldImages:-}" != "" ]]; then
            docker rmi --force $(docker images -a | grep "^ubnt/ucrm-billing" | awk '{print $3}')
        fi
    else
        echo "Docker not installed, skipping removal of docker container and images."
    fi

    if [[ -f "${UCRM_PATH}/docker-compose.yml" ]]; then
        echo "Removing ${UCRM_PATH}/docker-compose.yml."
        rm -f "${UCRM_PATH}/docker-compose.yml"
    fi

    if [[ -f "${UCRM_PATH}/docker-compose.migrate.yml" ]]; then
        echo "Removing ${UCRM_PATH}/docker-compose.migrate.yml."
        rm -f "${UCRM_PATH}/docker-compose.migrate.yml"
    fi

    if [[ -f "${UCRM_PATH}/docker-compose.version.yml" ]]; then
        echo "Removing ${UCRM_PATH}/docker-compose.version.yml."
        rm -f "${UCRM_PATH}/docker-compose.version.yml"
    fi

    if [[ -f "${UCRM_PATH}/elasticsearch.yml" ]]; then
        echo "Removing ${UCRM_PATH}/elasticsearch.yml."
        rm -f "${UCRM_PATH}/elasticsearch.yml"
    fi

    if [[ -f "${UCRM_PATH}/docker-compose.env" ]]; then
        echo -e "\n---\n"
        echo "File ${UCRM_PATH}/docker-compose.env is NOT removed and if you want to restore UCRM data in the future, you should do a backup."
        echo "Password for PostgreSQL database is saved in this file and you will not be able to use the current database data without it."
    fi
}

print_intro() {
    echo "+------------------------------------------------+"
    echo "| UCRM - Complete WISP Management Platform       |"
    echo "|                                                |"
    echo "| https://ucrm.ubnt.com/      (uninstaller v1.0) |"
    echo "+------------------------------------------------+"
    echo ""
}

print_intro
do_uninstall


update-cron.sh
#!/usr/bin/env bash

set -o errexit
set -o errtrace
set -o nounset
set -o pipefail
#set -o xtrace

UCRM_PATH="${UCRM_PATH:-}"
if [[ ! -d "${UCRM_PATH}" ]]; then
    UCRM_PATH=""
fi
if [[ "${UCRM_PATH}" = "" ]] && [[ "${BASH_SOURCE+x}" = "x" ]]; then
    UCRM_PATH="$(dirname "${BASH_SOURCE[0]}")"
fi
if [[ "${UCRM_PATH}" = "" ]]; then
    UCRM_PATH="."
fi

if (which realpath > /dev/null 2>&1); then
    UCRM_PATH="$(realpath "${UCRM_PATH}")"
else
    UCRM_PATH="$(cd "${UCRM_PATH}" > /dev/null && pwd)"
fi

export GITHUB_REPOSITORY="${GITHUB_REPOSITORY:-Ubiquiti-App/UCRM/master}"

if [[ ! -f "${UCRM_PATH}/docker-compose.yml" ]]; then
    echo "docker-compose.yml does not exist or is not readable in path ${UCRM_PATH}"

    exit 1
fi

get_ucrm_data_path() {
    if ! ( cat -vt "${UCRM_PATH}/docker-compose.yml" | grep -Eq "\.\/data\/ucrm:\/data" );
    then
        cat -vt "${UCRM_PATH}/docker-compose.yml" | grep -E "^      \- \/home\/.+:\/data$" -m 1 --color=never | awk ' {print $NF}' | awk -F: '{print $1}'
    else
        echo "${UCRM_PATH}/data/ucrm"
    fi
}
UCRM_DATA_PATH=$(get_ucrm_data_path)
UCRM_UPDATES_PATH="${UCRM_DATA_PATH}/updates"

if [[ ! -d "${UCRM_UPDATES_PATH}" ]]; then
    mkdir -p "${UCRM_UPDATES_PATH}"
fi

if [ ! -w "${UCRM_UPDATES_PATH}" ]; then
    echo "Cannot write into path ${UCRM_UPDATES_PATH}"
    exit 1
fi

UCRM_UPDATE_REQUESTED_FILE="${UCRM_UPDATES_PATH}/update_requested"
UCRM_UPDATE_RUNNING_FILE="${UCRM_UPDATES_PATH}/update_running"
UCRM_UPDATE_MAINTENANCE_FILE="${UCRM_UPDATES_PATH}/update_maintenance"
UCRM_UPDATE_LOG_FILE="${UCRM_UPDATES_PATH}/update.log"

if [[ -f "${UCRM_UPDATE_RUNNING_FILE}" ]]; then
    echo "$(date) --- Update process is already running."

    exit 1
fi

cleanup_old_logs() {
    (find "${UCRM_UPDATES_PATH}/logs" -maxdepth 1 -name 'update_*.log' -type f -printf "${UCRM_UPDATES_PATH}/logs/%f\n" | sort -r | tail -n +61 | xargs -I {} rm -- {})
}

backup_log_file() {
    declare toVersion="${1}"

    if [[ -f "${UCRM_UPDATE_LOG_FILE}" ]]; then
        if [[ ! -d "${UCRM_UPDATES_PATH}/logs" ]]; then
            mkdir -p "${UCRM_UPDATES_PATH}/logs"
        fi

        DATE=$(date +"%s")
        cp -f "${UCRM_UPDATE_LOG_FILE}" "${UCRM_UPDATES_PATH}/logs/update_${DATE}_${toVersion}.log"

        cleanup_old_logs
    fi
}

if [[ -f "${UCRM_UPDATE_REQUESTED_FILE}" ]]; then
    # redirect output to log file
    exec > "${UCRM_UPDATE_LOG_FILE}" 2>&1

    VERSION_TO_UPDATE=$(cat "${UCRM_UPDATE_REQUESTED_FILE}")

    echo "$(date) --- Initializing UCRM update to version ${VERSION_TO_UPDATE}."
    echo "$(date) --- Using update-cron.sh v1.0"

    touch "${UCRM_UPDATE_RUNNING_FILE}"
    trap 'rm -f "${UCRM_UPDATE_RUNNING_FILE}"; rm -f "${UCRM_UPDATE_MAINTENANCE_FILE}"; backup_log_file "${VERSION_TO_UPDATE}"; exit' INT TERM EXIT
    rm "${UCRM_UPDATE_REQUESTED_FILE}"

    echo "$(date) --- Downloading current updater."
    curl -o "${UCRM_PATH}/update.sh" "https://raw.githubusercontent.com/${GITHUB_REPOSITORY}/update.sh"
    export PATH="/usr/local/bin:$PATH"
    echo "$(date) --- Starting the update process."
    if ( bash "${UCRM_PATH}/update.sh" --version "${VERSION_TO_UPDATE}" --cron ); then
        echo "$(date) --- Update successful."
        rm -f "${UCRM_UPDATE_RUNNING_FILE}"
        rm -f "${UCRM_UPDATE_MAINTENANCE_FILE}"
        backup_log_file "${VERSION_TO_UPDATE}"

        exit 0
    else
        echo "$(date) --- Update failed."
        rm -f "${UCRM_UPDATE_RUNNING_FILE}"
        rm -f "${UCRM_UPDATE_MAINTENANCE_FILE}"
        backup_log_file "${VERSION_TO_UPDATE}"

        exit 1
    fi
fi

echo "$(date) --- UCRM update not requested."

exit 0


install-cloud.sh
#!/bin/bash

# This script will install ucrm on a cloud hosted server without any prompt
# Standard ports 80,81,443 will be used
# This script will append all variables in /tmp/cloud_conf (eg. SERVER_NAME) to docker-compose.env

INSTALL_CLOUD=true
CLOUD_CONF="/tmp/cloud_conf"

# Init username and password
UCRM_USERNAME="admin"
UCRM_PASSWORD="admin"
while [[ $# -gt 0 ]]
do
key="$1"
case "${key}" in
  --username)
    UCRM_USERNAME="$2"
    shift # past argument value
    ;;
  --password)
    UCRM_PASSWORD="$2"
    shift # past argument value
    ;;
  *)
    # unknown option
    ;;
esac
shift # past argument key
done
echo "UCRM_USERNAME=${UCRM_USERNAME}" >> "${CLOUD_CONF}"
echo "UCRM_PASSWORD=${UCRM_PASSWORD}" >> "${CLOUD_CONF}"

# Set up swap file
. /tmp/setup-swap.sh

# Install UCRM
. /tmp/install.sh


ucrm_install_debian.sh
#!/bin/sh

# linux user for ucrm docker containers
UCRM_USER="ucrm"
POSTGRES_PASSWORD=$(cat /dev/urandom | tr -dc "a-zA-Z0-9" | fold -w 48 | head -n 1);
SECRET=$(cat /dev/urandom | tr -dc "a-zA-Z0-9" | fold -w 48 | head -n 1);

DOCKER_CHANGE_8080_PORT=""
DOCKER_CHANGE_8081_PORT=""

which docker > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Download and install Docker"
	curl -fsSL https://get.docker.com/ | sh
fi

which docker > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Docker not installed. Please check previous logs. Aborting."
	exit 1
fi

which docker-compose > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Download and install Docker compose"
	curl -L https://github.com/docker/compose/releases/download/1.7.1/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
	chmod +x /usr/local/bin/docker-compose
fi

which docker-compose > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Docker compose not installed. Please check previous logs. Aborting."
	exit 1
fi

if [ -z "$(getent passwd $UCRM_USER)" ]; then
	echo "Creating user $UCRM_USER"
	adduser --disabled-password --gecos "" $UCRM_USER
	usermod -aG docker $UCRM_USER
fi

if [ ! -f /home/$UCRM_USER/docker-compose.yml ]; then
	echo "Downloading docker compose files"
	curl -o /home/$UCRM_USER/docker-compose.yml https://raw.githubusercontent.com/U-CRM/billing/master/docker-compose.yml
	curl -o /home/$UCRM_USER/docker-compose.env https://raw.githubusercontent.com/U-CRM/billing/master/docker-compose.env

	echo "Replacing path in docker compose"
	sed -i -e "s/#volumes:/volumes:/g" /home/$UCRM_USER/docker-compose.yml
	sed -i -e "s/#  \- \/home\/docker\/ucrm\/postgres:\/var\/lib\/postgresql\/data/  - \/home\/$UCRM_USER\/postgres:\/var\/lib\/postgresql\/data/g" /home/$UCRM_USER/docker-compose.yml
	sed -i -e "s/#  \- \/home\/docker\/ucrm:\/data/  - \/home\/$UCRM_USER\/data:\/data/g" /home/$UCRM_USER/docker-compose.yml

	echo "Replacing env in docker compose"
	sed -i -e "s/POSTGRES_PASSWORD=ucrmdbpass1/POSTGRES_PASSWORD=$POSTGRES_PASSWORD/g" /home/$UCRM_USER/docker-compose.env
	sed -i -e "s/SECRET=changeThisSecretKey/SECRET=$SECRET/g" /home/$UCRM_USER/docker-compose.env

	while true; do
		read -r -p "Do you want UCRM to be accessible on port 80? (Yes: recommended for most users, No: will set 8080 as default) [Y/n]: " DOCKER_CHANGE_8080_PORT

		case $DOCKER_CHANGE_8080_PORT in
			[yY][eE][sS]|[yY])
				sed -i -e "s/- 8080:80/- 80:80/g" /home/$UCRM_USER/docker-compose.yml
				echo "U CRM will start at 80 port"
				echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				echo "SERVER_PORT=80" >> /home/$UCRM_USER/docker-compose.env
				break;;
			[nN][oO]|[nN])
				echo "U CRM will start at 8080 port. If you will change it, edit your docker-compose.yml in $UCRM_USER home direcotry."
				echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				echo "SERVER_PORT=8080" >> /home/$UCRM_USER/docker-compose.env
				break;;
			*)
				;;
		esac
	done

	while true; do
		read -r -p "Do you want UCRM suspend page to be accessible on port 81? (Yes: recommended for most users, No: will set 8081 as default) [Y/n]: " DOCKER_CHANGE_8081_PORT

		case $DOCKER_CHANGE_8081_PORT in
			[yY][eE][sS]|[yY])
				sed -i -e "s/- 8081:81/- 81:81/g" /home/$UCRM_USER/docker-compose.yml
				echo "U CRM suspend page will start at 81 port"
				echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				echo "SERVER_SUSPEND_PORT=81" >> /home/$UCRM_USER/docker-compose.env
				break;;
			[nN][oO]|[nN])
				echo "U CRM suspend page will start at 8081 port. If you will change it, edit your docker-compose.yml in $UCRM_USER home direcotry."
				echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				echo "SERVER_SUSPEND_PORT=8081" >> /home/$UCRM_USER/docker-compose.env
				break;;
			*)
				;;
		esac
	done
fi

echo "Downloading docker images"
cd /home/$UCRM_USER && /usr/local/bin/docker-compose pull

echo "Starting docker images"
cd /home/$UCRM_USER && /usr/local/bin/docker-compose up -d && /usr/local/bin/docker-compose ps

exit 0


ucrm_install_ubuntu.sh
#!/bin/bash

# linux user for ucrm docker containers
UCRM_USER="ucrm"
POSTGRES_PASSWORD=$(cat /dev/urandom | tr -dc "a-zA-Z0-9" | fold -w 48 | head -n 1)
SECRET=$(cat /dev/urandom | tr -dc "a-zA-Z0-9" | fold -w 48 | head -n 1)

DOCKER_CHANGE_8080_PORT=""
DOCKER_CHANGE_8081_PORT=""

which docker > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Download and install Docker"
	curl -fsSL https://get.docker.com/ | sh
fi

which docker > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Docker not installed. Please check previous logs. Aborting."
	exit 1
fi

which docker-compose > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Download and install Docker compose"
	sudo curl -L https://github.com/docker/compose/releases/download/1.7.1/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
	sudo chmod +x /usr/local/bin/docker-compose
fi

which docker-compose > /dev/null 2>&1

if [ $? = 1 ]; then
	echo "Docker compose not installed. Please check previous logs. Aborting."
	exit 1
fi

if [ -z "$(getent passwd $UCRM_USER)" ]; then
	echo "Creating user $UCRM_USER"
	sudo adduser --disabled-password --gecos "" $UCRM_USER
	sudo usermod -aG docker $UCRM_USER
fi

if [ ! -f /home/$UCRM_USER/docker-compose.yml ]; then
	echo "Downloading docker compose files"
	sudo -u $UCRM_USER -H curl -o /home/$UCRM_USER/docker-compose.yml https://raw.githubusercontent.com/U-CRM/billing/master/docker-compose.yml
	sudo -u $UCRM_USER -H curl -o /home/$UCRM_USER/docker-compose.env https://raw.githubusercontent.com/U-CRM/billing/master/docker-compose.env

	echo "Replacing path in docker compose"
	sudo -u $UCRM_USER -H sed -i -e "s/#volumes:/volumes:/g" /home/$UCRM_USER/docker-compose.yml
	sudo -u $UCRM_USER -H sed -i -e "s/#  \- \/home\/docker\/ucrm\/postgres:\/var\/lib\/postgresql\/data/  - \/home\/$UCRM_USER\/postgres:\/var\/lib\/postgresql\/data/g" /home/$UCRM_USER/docker-compose.yml
	sudo -u $UCRM_USER -H sed -i -e "s/#  \- \/home\/docker\/ucrm:\/data/  - \/home\/$UCRM_USER\/data:\/data/g" /home/$UCRM_USER/docker-compose.yml

	echo "Replacing env in docker compose"
	sudo -u $UCRM_USER -H sed -i -e "s/POSTGRES_PASSWORD=ucrmdbpass1/POSTGRES_PASSWORD=$POSTGRES_PASSWORD/g" /home/$UCRM_USER/docker-compose.env
	sudo -u $UCRM_USER -H sed -i -e "s/SECRET=changeThisSecretKey/SECRET=$SECRET/g" /home/$UCRM_USER/docker-compose.env

	while true; do
		read -r -p "Do you want UCRM to be accessible on port 80? (Yes: recommended for most users, No: will set 8080 as default) [Y/n]: " DOCKER_CHANGE_8080_PORT

		case $DOCKER_CHANGE_8080_PORT in
			[yY][eE][sS]|[yY])
				sudo -u $UCRM_USER -H sed -i -e "s/- 8080:80/- 80:80/g" /home/$UCRM_USER/docker-compose.yml
				echo "UCRM will start at port 80"
				sudo -u $UCRM_USER -H echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				sudo -u $UCRM_USER -H echo "SERVER_PORT=80" >> /home/$UCRM_USER/docker-compose.env
				break;;
			[nN][oO]|[nN])
				echo "UCRM will start at port 8080. If you want to change it, edit your docker-compose.yml in $UCRM_USER home directory."
				sudo -u $UCRM_USER -H echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				sudo -u $UCRM_USER -H echo "SERVER_PORT=8080" >> /home/$UCRM_USER/docker-compose.env
				break;;
			*)
				;;
		esac
	done

	while true; do
		read -r -p "Do you want UCRM suspend page to be accessible on port 81? (Yes: recommended for most users, No: will set 8081 as default) [Y/n]: " DOCKER_CHANGE_8081_PORT

		case $DOCKER_CHANGE_8081_PORT in
			[yY][eE][sS]|[yY])
				sudo -u $UCRM_USER -H sed -i -e "s/- 8081:81/- 81:81/g" /home/$UCRM_USER/docker-compose.yml
				echo "UCRM suspend page will start at port 81"
				sudo -u $UCRM_USER -H echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				sudo -u $UCRM_USER -H echo "SERVER_SUSPEND_PORT=81" >> /home/$UCRM_USER/docker-compose.env
				break;;
			[nN][oO]|[nN])
				echo "UCRM suspend page will start at port 8081. If you want to change it, edit your docker-compose.yml in $UCRM_USER home directory."
				sudo -u $UCRM_USER -H echo "#used only in instalation" >> /home/$UCRM_USER/docker-compose.env
				sudo -u $UCRM_USER -H echo "SERVER_SUSPEND_PORT=8081" >> /home/$UCRM_USER/docker-compose.env
				break;;
			*)
				;;
		esac
	done
fi

echo "Downloading docker images"
cd /home/$UCRM_USER && sudo -u $UCRM_USER -H /usr/local/bin/docker-compose pull

echo "Starting docker images"
cd /home/$UCRM_USER && sudo -u $UCRM_USER -H /usr/local/bin/docker-compose up -d && /usr/local/bin/docker-compose ps

exit 0
